netbench/README.md

```diff
@@ -62,16 +78,50 @@ pub fn scenario(config: Config) -> Scenario {
 }
 ```
 
-This scenario generates a json file of instructions. These instructions are protocol and language independent, which means they can easily be executed by a ["netbench driver"](./netbench-driver/), written in any language or runtime. Implemented drivers include:
+This scenario generates a json file of instructions. These instructions are protocol and language independent, which means they can easily be executed by a ["netbench driver"](./netbench-driver/), written in any language or runtime.
+
+### netbench-driver
+Netbench drivers are responsible for executing netbench scenarios. Each transport protocol has a `client` and `server` implementation. Each of these implementations is a self-container binary that consumes a `scenario.json` file. Implemented drivers include:
 
 * `TCP`
 * [`native-tls`](https://crates.io/crates/native-tls)
     * OpenSSL on Linux
     * Secure Transport on macOS
     * SChannel on Windows
 * `s2n-quic`
-* `s2n-tls` (coming soon)
+* `s2n-tls`
+
+### netbench-collector
+Driver metrics are collected with the [`netbench-collector`](./netbench-collector/) utility. There are two implementation of this available - a generic utility and a bpftrace utility. The generic utility uses the `proc fs` to gather information about the process, while the `bpftrace` implementation is able to collect a wider variety of statistics through ebpf probes.
+
+The collector binary takes a `netbench-driver` as an argument. The driver binary is spawned as a child process. The collector will continuously gather metrics from the driver and emit those metrics to `stdout`.
+
+### netbench-cli
+`netbench-cli` has two commands.
 
-Driver metrics are collected with the [`netbench-collector`](./netbench-collector/) utility. Reports are then generated for the collected metrics with the [`cli`](./netbench-cli/).
+`report` is used to generate individual `.json` reports, and is not expected to be useful in isolation.
```

Comment:
> and is not expected to be useful in isolation.

I'm not sure what you mean here

Reply:
Ya, this could be worded a bit better. The meaning is that people aren't expected to directly use the`report` command since it doesn't output particularly human readable information, and instead they should be using the `report-tree` command.

It also might be the case that I just haven't figured out the nice way to visualize the report `.json`s by themselves.